File: Laborator/CSharp/AgentieTurism/Tests/FlightRepositoryTest.cs
using System;
using System.Collections.Generic;
using System.IO;
using NUnit.Framework;
using AgentieTurism.Models;
using AgentieTurism.Repository;

namespace AgentieTurism.Tests
{
    [TestFixture]
    public class FlightRepositoryTest
    {
        private static IFlightRepository flightRepo;
        private static Flight flight;

        [OneTimeSetUp]
        public void Setup()
        {
            //var props = new Dictionary<string, string>();
            //foreach (var line in File.ReadLines("bd.config"))
            //{
            //    var parts = line.Split('=');
            //    if (parts.Length == 2)
            //        props[parts[0].Trim()] = parts[1].Trim();
            //}

            //flightRepo = new FlightDbRepository(props);
            flightRepo = new FlightDbRepository();
        }

        [Test, Order(1)]
        public void TestAddFlight()
        {
            flight = new Flight("London", new DateTime(2025, 5, 10, 15, 0, 0), "Heathrow", 200);
            flightRepo.Add(flight);
            Assert.That(flight.Id, Is.Not.Null);
        }

        [Test, Order(2)]
        public void TestFindFlightById()
        {
            var foundFlight = flightRepo.FindOne(flight.Id);
            Assert.That(foundFlight, Is.Not.Null);
            Assert.That(foundFlight.Destination, Is.EqualTo("London"));
        }

        [Test, Order(3)]
        public void TestUpdateFlight()
        {
            flight.AvailableSeats = 150;
            flightRepo.Update(flight.Id, flight);
            var updatedFlight = flightRepo.FindOne(flight.Id);
            Assert.That(updatedFlight.AvailableSeats, Is.EqualTo(150));
        }

        [Test, Order(4)]
        public void TestFindAllFlights()
        {
            var flights = flightRepo.FindAll();
            Assert.That(flights, Is.Not.Empty);
        }

        [Test, Order(5)]
        public void TestDeleteFlight()
        {
            flightRepo.Delete(flight.Id);
            Assert.That(flightRepo.FindOne(flight.Id), Is.Null);
        }
    }
}

File: Laborator/Java/AgentieTurism/src/test/java/FlightRepositoryTest.java
import org.junit.jupiter.api.*;
import ro.mpp2024.domain.Flight;
import ro.mpp2024.repository.FlightDBRepository;
import ro.mpp2024.repository.FlightRepository;

import java.io.FileReader;
import java.io.IOException;
import java.time.LocalDateTime;
import java.util.List;
import java.util.Properties;

import static org.junit.jupiter.api.Assertions.*;

@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
public class FlightRepositoryTest {
    private static FlightRepository flightRepo;
    private static Flight flight;

    @BeforeAll
    static void setup() {
        Properties props = new Properties();
        try {
            props.load(new FileReader("bd.config"));
        } catch (IOException e) {
            fail("Cannot find bd.config: " + e);
        }
        flightRepo = new FlightDBRepository(props);
    }

    @Test
    @Order(1)
    void testAddFlight() {
        flight = new Flight("London", LocalDateTime.of(2025, 5, 10, 15, 0), "Heathrow", 200);
        flightRepo.add(flight);
        assertNotNull(flight.getId());
    }

    @Test
    @Order(2)
    void testFindFlightById() {
        Flight foundFlight = flightRepo.findOne(flight.getId());
        assertNotNull(foundFlight);
        assertEquals("London", foundFlight.getDestination());
    }

    @Test
    @Order(3)
    void testUpdateFlight() {
        flight.setAvailableSeats(150);
        flightRepo.update(flight.getId(), flight);
        Flight updatedFlight = flightRepo.findOne(flight.getId());
        assertEquals(150, updatedFlight.getAvailableSeats());
    }

    @Test
    @Order(4)
    void testFindAllFlights() {
        List<Flight> flights = (List<Flight>) flightRepo.findAll();
        assertFalse(flights.isEmpty());
    }

    @Test
    @Order(5)
    void testDeleteFlight() {
        flightRepo.delete(flight.getId());
        assertNull(flightRepo.findOne(flight.getId()));
    }
}
